### PR TITLE
fix(docs/typescript): provide example how to use middlewares with #1565

### PR DESCRIPTION
**Pull request by holgergp**
## Related Issues
#1565 

Fixes #.

## Summary
Providing an example of using middlewares with vanilla stores using typescript

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: docs/guides/typescript.md [hunk 1]
```diff
@@ -467,6 +467,22 @@ export const BearComponent = () => {
...
+
+export const vanillaBearStore = createStore<BearState>()(
+  devtools((set, getState) => ({
+    ...initialBearState,
+    increase: (by) => set((state) => ({ bears: state.bears + by })),
+  }))
+)
+```
+
+For the extra parantheses please see [this](https://github.com/pmndrs/zustand/blob/main/docs/guides/typescript.md#basic-usage)
```
Comment (sewera) [inline docs/guides/typescript.md:484]:
```suggestion
For more information about why there are extra parentheses,
please see the [Basic usage section](#basic-usage).
```

**Review by sewera — CHANGES_REQUESTED** [review]
I believe linking to the section, so that it's available both on Github, and in the rendered <https://docs.pmnd.rs/zustand/guides/typescript#basic-usage> is a better idea.

**Review by sewera — APPROVED** [review]
Thanks for commit mention :)

**Comment by holgergp** [comment]
@sewera NP :)
This was the inline GH "apply suggestion" workflow. Apparently it generates a co-author. 🤷  Which is nice! 😀

**Pull request merged**